### PR TITLE
feat: add ETH gas fees price estimation utility methods to @t3rn/sdk

### PR DESCRIPTION
**Pull request by ahkohd**
# Summary of changes

**_Summarize_** the description of your changes.

If it fixes a bug or creates a feature, link the issue.

# Acceptance Checklist

What types of changes does your code introduce?
_Put an `x` in the boxes that apply_

## Dependencies

- [ ] Any dependent changes have been merged and published in downstream modules
- [ ] Any teams which manage the dependencies have been notified of the breaking changes

## Bug (non-breaking change which fixes an issue):

- [x] Have you **_written tests_** to protect against future occurrences of the bug?

## Feature (non-breaking change which adds functionality)

- [x] Have you **_seen it working_** in a development environment?
- [x] Have you **_written tests_** for **_happy_** and **_unhappy_** paths?
- [ ] Have you implemented this feature as per the **_issue acceptance criteria_**?

### Substrate

If this change involves substrate, have you remembered:

- [ ] Storage Migration?
- [ ] RPC methods?
- [ ] Chainspec, both JSON specs & the module?
- [ ] Runtime versioning?
- [ ] Benchmarks?

## Breaking change (a feature that would cause existing functionality not to work as expected)

- [ ] Is the breaking change **_documented_**?
- [ ] Are your commits fully representative of the change?
- [ ] Has any previous functionality been **_deprecated_** and versioned?

## CI/CD

- [ ] If introducing new actions, have you **_looked at the action code_** for anything spurious?
- [ ] Have you written **_custom scripts_** for things that could be actions already on the marketplace?
- [ ] If introducing new actions, have you **_pegged a static version_**?

## Documentation Update

- [ ] Have you checked other documentation, such as the docs repository?
- [ ] If updating script documentation, have you **_tested it on both environments_**?

## Further comments

Feel free to explain in further detail your choices if this is a significant change.

## Screenshots (Please demonstrate this working in PolkadotJS)

Closes https://github.com/t3rn/t3rn/issues/1163

### Summary by OpenAI

Release Notes:

- New Feature: Added a new module `ethGasFeeEstimationUtils` to the SDK for estimating Ethereum gas fees.
- New Feature: Introduced utility methods for estimating ETH gas fees using the CoinGecko API.
- New Feature: Added utility methods for estimating ETH gas fees in the `@t3rn/sdk` package.
- New Feature: Added a new command "estimate-fees" to the CLI, which estimates the fees for executing an Sfx.
- Refactor: Updated import statements, function signatures, and method calls in various files.
- Test: Added test cases for the `estimateMaxReward` function in the `price-estimation` module.

> "Gas fees estimated,
> SDK empowered with new features,
> CLI commands expanded,
> Bugs squashed, tests strengthened,
> A release that brings efficiency and pleasure."

### CODE REVIEW

File: client/packages/cli/src/commands/estimate.ts [hunk 1]
```diff
@@ -0,0 +1,120 @@
...
+const estimateFees = async (
+  tag: string,
+  sideEffect: SideEffect,
+  speedMode: SpeedMode
+) => {
+  const withArgs = (fn: (...args: unknown[]) => unknown) =>
+    fn(tag, sideEffect, speedMode)
+
+  switch (sideEffect.target) {
+    case "roco":
```
Comment (MaciejBaj) [inline client/packages/cli/src/commands/estimate.ts:63]: "polk", "ksma"

File: client/packages/sdk/src/price-estimation/eth.ts [hunk 2]
```diff
@@ -0,0 +1,56 @@
...
+  rapid: number;
+  fast: number;
+  standard: number;
+  slow: number;
+  timestamp: number;
+  priceUsd: number;
+};
+
+export const getGasPrice = async () => {
+  const endpoint = "https://beaconcha.in/api/v1/execution/gasnow";
```
Comment (MaciejBaj) [inline client/packages/sdk/src/price-estimation/eth.ts:29]: maybe select for 2 gateways linked to 2 different APIs for Ethereum + Sepolia?

Comment (ahkohd) [reply client/packages/sdk/src/price-estimation/eth.ts]: Fixed, see commit [9825cee](https://github.com/t3rn/t3rn/pull/1091/commits/9825ceeba5c279173329a3755d4e4d83bb331890).

File: client/packages/cli/src/commands/estimate.ts (hunk 1 above)
Comment (ahkohd) [reply client/packages/cli/src/commands/estimate.ts:63]: Fixed, refer to commit [853a45d](https://github.com/t3rn/t3rn/pull/1091/commits/853a45d48f5b9af932bc6b7bd2fbed93c87e34fa).

File: client/packages/yarn.lock [hunk 3]
```diff
@@ -0,0 +1,373 @@
+# THIS IS AN AUTOGENERATED FILE. DO NOT EDIT THIS FILE DIRECTLY.
```
Comment (3h4x) [inline client/packages/yarn.lock:1]: Why this file was added?

File: node/t0rn-parachain/Cargo.toml [hunk 4]
```diff
@@ -1,9 +1,9 @@
 [package]
```
Comment (3h4x) [inline node/t0rn-parachain/Cargo.toml:1]: This PR shouldnt introduce changes to cargo.toml

File: runtime/t0rn-parachain/Cargo.toml
Comment (3h4x) [inline runtime/t0rn-parachain/Cargo.toml:1]: This PR shouldnt introduce changes to cargo.toml

File: client/packages/executor/package.json [hunk 5]
```diff
@@ -5,16 +5,17 @@
...
+    "dev": "yarn build && LOG_LEVEL=debug EXECUTOR=example node dist/src/main.js | npx pino-pretty -c -S",
     "lint": "eslint --ext .ts src",
     "lint:report": "eslint src --ext .ts --output-file eslint_report.json --format json",
     "export": "LOG_LEVEL=debug yarn start >> logger.log",
     "fmt": "prettier --write src config tests",
     "build-docs": "npx typedoc && cp -R docs ../../../docs/executor",
-    "test": "mocha -r ts-node/register 'tests/**/*.ts'",
-    "test:attestion": "mocha -r ts-node/register 'tests/**/attest*.ts'",
-    "test-single": "mocha -r ts-node/register"
+    "test": "yarn build && mocha 'dist/tests/**/*.js'",
```
Comment (3h4x) [inline client/packages/executor/package.json:16]: test, start, etc should run yarn build
build should build

**Review by 3h4x — CHANGES_REQUESTED** [review]
CLI doesn't work for me.
Please run make clean, then make and try it again

Screenshot 2023-07-14 at 15 20 57

File: client/packages/sdk/src/price-estimation/__tests__/price-estimation.test.ts [hunk 6]
```diff
@@ -0,0 +1,23 @@
+import { describe, expect, it } from "@jest/globals";
```
Comment (3h4x) [inline client/packages/sdk/src/price-estimation/__tests__/price-estimation.test.ts:1]: This test is not mocked so each run will trigger API call to coingecko

File: client/packages/sdk/src/price-estimation/__tests__/eth.test.ts [hunk 7]
```diff
@@ -0,0 +1,99 @@
...
+      } catch (e) {
+        expect(e).toEqual(
+          new Error("Failed to fetch gas price. ERROR_STATUS: 500")
+        );
+      }
+    });
+  });
+
+  describe("estimateGasFee", () => {
+    it("should get the estimate gas fee for a transfer", async () => {
```
Comment (3h4x) [inline client/packages/sdk/src/price-estimation/__tests__/eth.test.ts:73]: Please test all speed modes

File: client/packages/yarn.lock (hunk 3 above)
Comment (ahkohd) [reply client/packages/yarn.lock]: I made an error, but it has been corrected. Please review the commit 9185d4080308838736851d385e28a5303155f8dc.

File: client/packages/executor/package.json (hunk 5 above)
Comment (ahkohd) [reply client/packages/executor/package.json]: Okay, see the commit [fc57e2f](https://github.com/t3rn/t3rn/pull/1091/commits/fc57e2f4c75aed2b08eb3889f0159fc368b0bbac).

File: runtime/t0rn-parachain/Cargo.toml
Comment (ahkohd) [reply runtime/t0rn-parachain/Cargo.toml:1]: Yes, this was a mistake; see commit [27f0faa](https://github.com/t3rn/t3rn/pull/1091/commits/27f0faa9bf7b6d7e9e42c51c4b560f939265cc88).

**Comment by ahkohd** [comment]
> * We should have additional documentation to address this CLI feature.
> * Output is not clear
> 
> Please revisit

I have updated the README.md file, See commit 4352a3b45fa8148a1051dc19086c42cc1106d1af.

File: client/packages/sdk/src/price-estimation/__tests__/price-estimation.test.ts (hunk 6 above)
Comment (ahkohd) [reply client/packages/sdk/src/price-estimation/__tests__/price-estimation.test.ts]: Fixed, refer to commit [659bacf](https://github.com/t3rn/t3rn/pull/1091/commits/659bacf98fbb5ec05ae91e992312e82db7cdaf79).

File: client/packages/sdk/src/price-estimation/__tests__/eth.test.ts (hunk 7 above)
Comment (ahkohd) [reply client/packages/sdk/src/price-estimation/__tests__/eth.test.ts:73]: Fixed, refer to commit [0fd588d](https://github.com/t3rn/t3rn/pull/1091/commits/0fd588dfbcb1b371e9b60d1415e20932aa3e1ab4).

**Review by 3h4x — APPROVED** [review]
LGTM

**Pull request merged**